src/bootstrap/src/core/build_steps/gcc.rs

```diff
@@ -0,0 +1,36 @@
+//! Retrieves information related to GCC backend.
+
+use crate::builder::Cargo;
+
+/// This retrieves the gccjit sha we use.
+pub(crate) fn detect_gccjit_sha() -> String {
+    // FIXME: This is absolutely ugly. Please find another way to do that.
+    let content =
+        include_str!("../../../../ci/docker/host-x86_64/dist-x86_64-linux/build-gccjit.sh");
+    if let Some(commit) = content
+        .lines()
+        .filter_map(|line| line.strip_prefix("GIT_COMMIT=\""))
+        .filter_map(|commit| commit.strip_suffix("\""))
+        .next()
+    {
+        if !commit.is_empty() {
+            return commit.to_string();
+        }
+    }
+
+    eprintln!("error: could not find commit hash for downloading libgccjit");
+    panic!();
+}
+
+pub fn set_gccjit_library_path(config: &crate::Config, cargo: &mut Cargo) {
+    if config.gcc_download_gccjit {
+        let libgccjit_path = config.libgccjit_folder(&detect_gccjit_sha());
+        let mut library_path = std::env::var("LD_LIBRARY_PATH").unwrap_or_default();
+        if !library_path.is_empty() {
+            library_path.push(':');
+        }
+        library_path.push_str(&libgccjit_path.display().to_string());
```

Comment:
Hm. Is this the right way around? Should this be prefixing the path instead so it is preferred over other libgccjits that may be encountered?

Reply:
I thought that if the user specificied a `LD_LIBRARY_PATH`, they might want to have it used over the one provided by bootstrap.

Reply:
Hmm, okay! It seems somewhat arbitrary either way, I guess.

Reply:
Well in this case I now rely on what bootstrap is already doing following suggestions from @onur-ozkan so not an issue anymore. ;)